docs/faq.md

```diff
@@ -109,6 +109,46 @@ commands =
 `tox` will not do any install. Poetry installs all the dependencies and the current package an editable mode.
 Thus, tests are running against the local files and not the built and installed package.
 
+#### Usecase #4
+```ini
+[tox]
+isolated_build = true
+
+[testenv]
+allowlist_externals = poetry
+commands_pre =
+    poetry install --no-root --sync --with test
+commands =
+    pytest tests/
+
+[testenv:style]
+skip_install = true
+commands_pre =
+    poetry install --no-root --sync --with style
+commands =
+    flake8 src/
+    black --check src/
+
+[testenv:typing]
+commands_pre =
+    poetry install --no-root --sync --with typing
+commands =
+    mypy src/
```

Comment:
Typechecking should be better done against the editable install. So we need `skip_install = true` and the `--no-root` parameter must be removed.

Reply:
I've had real trouble with this under other toolchains (e.g. setuptools) because you need the full set of runtime dependencies installed into the type-checking environment. But I think what you're describing works for `poetry` because we are doing an explicit `poetry install` step. I'll test it out and if it works I can make the update.

Reply:
Confirmed that this works as desired and updated. I'm going to mark this as resolved.
I also added a sentence or two below to better explain the distinction between the `style` and `typing` testenvs, with this change.